### ISSUE TITLE
Backend files / folders show up alongside with the items created

### ISSUE THREAD

**Issue by Gaming4LifeDE** [spheras/desktopfolder#260]
### Note

I am maintaining desktopfolder on Solus. I encountered this while I tested it in order to push the update.

### Description
I created a folder, a panel and a note. The files representing the items show up on the desktop.
Upon running desktopfolder desktop icon display should be automatically disabled, if the desktop environment supports it. This also glitches the mouse selection out a bit (stuck in select mode)

### Steps to reproduce
1. create a folder / panel / etc.

### Expected result
Only the created item shows up.

### Actual result
The files / folder representing the items show up too.
[Image showing the bug](https://i.imgur.com/5DIuPWO.png)

### Possible fix
Turn off desktop icon display on desktop environments which support the feature.

### System Information
- OS: Solus 4.0 Fortitude (unstable), Solus 3.999 (stable), Budgie Edition
- Desktop Folder version: 1.1.1

**Comment by spheras** [comment]
Hi @Gaming4LifeDE, I'm a bit lost about this issue. The icons displayed are managed by a different application? (nautilus maybe?)  I mean, desktopfolder should not show those icons, I see there panels and notes and their respective files, so I imagine you have 2 managers for the desktop actives.  Please, could you confirm this?

If that is true, you mean to deactivate from desktopfolder the other desktop manager active?

**Comment by Gaming4LifeDE** [comment]
Yes, it's managed by Nautilus. Can you disable it by checking if Nautilus
has the option enabled when starting desktopfolder and disable the option
if necessary?


On 07.03.2019 at 11:53, <José Amuedo Salmerón <notifications@github.com>>
wrote:

Hi @Gaming4LifeDE <https://github.com/Gaming4LifeDE>, I'm a bit lost about
this issue. The icons displayed are managed by a different application?
(nautilus maybe?) I mean, desktopfolder should not show those icons, I see
there panels and notes and their respective files, so I imagine you have 2
managers for the desktop actives. Please, could you confirm this?

If that is true, you mean to deactivate from desktopfolder the other
desktop manager active?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/spheras/desktopfolder/issues/260#issuecomment-470480288>,
or mute the thread

**Comment by spheras** [comment]
Do you know how could I deactivate nautilus desktop? maybe @fossfreedom could help a bit here. On the other hand, I'm not sure if this is the correct way, I mean, deactivate a different desktop manager from the application... and seems something too specific, don't you think? I am open to discussing it.

**Comment by fossfreedom** [comment]
I wouldn't try to deactivate another desktop handler.  A user should be making a conscious decision "I want to use this desktop handler".

There isn't an easy way to deactivate nautilus - once nautilus 3.30 and later is available, there is no dconf key to fiddle with - and trying to test keys that don't exist causes crash issues.

**Comment by Gaming4LifeDE** [comment]
I have no idea what's wrong with the gnome devs, trimming features all the
time. It sure gets annoying.

I'd say if you install desktopfolder as a user you kind of made a conscious
decision, otherwise why would you install it in the first place?


On 07.03.2019 at 15:53, <David Mohammed <notifications@github.com>> wrote:

I wouldn't try to deactivate another desktop handler. A user should be
making a conscious decision "I want to use this desktop handler".

There isn't an easy way to deactivate nautilus - once nautilus 3.30 and
later is available, there is no dconf key to fiddle with - and trying to
test keys that don't exist causes crash issues.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/spheras/desktopfolder/issues/260#issuecomment-470556174>,
or mute the thread

**Comment by spheras** [comment]
I see some problematic points to consider:

1. Real possibility to deactivate nautilus from code (it seems very difficult, we need specific info about this possibility)
1. Probably to perform those changes DF will need root permissions 
1. This seems a very specific issue to nautilus, What about others desktop handlers?

To be honest, I don't like the idea to modify external applications from DesktopFolder.. but, not sure, maybe asking the user before, or something similar could be enough (thinking loud)...  This seems something too specific, and I'm not sure how to manage it in an abstract way.  Maybe it could be a lot easier just to detect that situation and notify to the user. Perhaps, in the notification, link to a web with some info about the manual deactivation of Nautilus Desktop Handler... not sure.

I'm tagging the issue as an improvement and will see in the future.  Any information/opinion is welcome.
